omni-relayer/src/workers/near.rs

```diff
@@ -249,84 +287,136 @@ pub async fn claim_fee(
 
         for (key, event) in events {
             if let Ok(fin_transfer) = serde_json::from_str::<FinTransfer>(&event) {
-                let vaa = utils::evm::get_vaa_from_evm_log(
-                    connector.clone(),
-                    fin_transfer.chain_kind,
-                    fin_transfer.tx_logs,
-                    &config,
-                )
-                .await;
-
-                if vaa.is_none() {
-                    let Ok(light_client_latest_block_number) =
-                        utils::near::get_eth_light_client_last_block_number(
-                            &config,
-                            &jsonrpc_client,
-                        )
-                        .await
-                    else {
-                        warn!("Failed to get eth light client last block number");
-                        continue;
-                    };
-
-                    if fin_transfer.block_number > light_client_latest_block_number {
-                        tokio::time::sleep(tokio::time::Duration::from_secs(
-                            utils::redis::SLEEP_TIME_AFTER_EVENTS_PROCESS_SECS,
-                        ))
-                        .await;
-                        continue;
-                    }
-                }
-
-                handlers.push(tokio::spawn({
-                    let config = config.clone();
-                    let mut redis_connection = redis_connection.clone();
-                    let connector = connector.clone();
+                if let FinTransfer::Evm {
+                    chain_kind,
+                    block_number,
+                    log,
+                    tx_logs,
+                } = fin_transfer
+                {
+                    info!("Trying to process FinTransfer log on {:?}", chain_kind);
+
+                    let vaa = utils::evm::get_vaa_from_evm_log(
+                        connector.clone(),
+                        chain_kind,
+                        tx_logs,
+                        &config,
+                    )
+                    .await;
+
+                    if vaa.is_none() {
+                        let Ok(light_client_latest_block_number) =
+                            utils::near::get_eth_light_client_last_block_number(
```

Comment:
You should check chain_kind here. It can be not Eth

Reply:
Fixed in https://github.com/Near-One/omni-bridge/pull/153/commits/66e03903425f7b9dabba7f744ad838f2cf2aa803

Reply:
We can use wormhole for Eth...

Reply:
Refixed in https://github.com/Near-One/omni-bridge/pull/153/commits/4346b420b085f9aa108e6b17655444ef25c853ea 😅